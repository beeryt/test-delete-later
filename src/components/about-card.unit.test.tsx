import React from 'react'
import { render } from '@testing-library/react'

import * as gatsby from 'gatsby'
const useStaticQuery = jest.spyOn(gatsby, 'useStaticQuery')
jest.mock(
  'gatsby-image',
  () =>
    function Img() {
      return <img />
    }
)

import AboutCard from './about-card'

describe('AboutCard', () => {
  it('renders correctly with mocked query', () => {
    useStaticQuery.mockImplementationOnce(() => {
      return { file: { childImageSharp: { fluid: {} } } }
    })
    render(<AboutCard />)
    expect(document.body).toMatchSnapshot()
  })

  it('renders correctly with missing query', () => {
    render(<AboutCard />)
    expect(document.body).toMatchSnapshot()
  })
})
