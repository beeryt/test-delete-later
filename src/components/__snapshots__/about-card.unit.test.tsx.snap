// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutCard renders correctly with missing query 1`] = `
<body>
  <div>
    <div
      class="card"
    >
      <div
        class="card-image"
      >
        <figure
          class="image"
        >
          <img
            alt="Placeholder image"
            src="https://bulma.io/images/placeholders/1280x960.png"
            title=""
          />
        </figure>
      </div>
      <div
        class="card-content"
      >
        <div
          class="media"
        >
          <div
            class="media-content"
          >
            <p
              class="title is-4"
            >
              Carl B. Smiley
            </p>
            <p
              class="subtitle is-6"
            >
              @linux_pangolin
            </p>
          </div>
        </div>
        <div
          class="level is-mobile"
        >
          <a
            aria-label="GitHub"
            class="level-item"
            href="https://github.com/beeryt"
          >
            <span
              class="icon"
            >
              <i
                class="fab fa-github"
              />
            </span>
          </a>
          <a
            aria-label="LinkedIn"
            class="level-item"
            href="https://www.linkedin.com/in/carl-b-smiley"
          >
            <span
              class="icon"
            >
              <i
                class="fab fa-linkedin"
              />
            </span>
          </a>
          <a
            aria-label="StackExchange"
            class="level-item"
            href="https://stackexchange.com/users/13595322/linux-pangolin"
          >
            <span
              class="icon"
            >
              <i
                class="fab fa-stack-overflow"
              />
            </span>
          </a>
          <a
            aria-label="Twitter"
            class="level-item"
            href="https://twitter.com/linux_pangolin"
          >
            <span
              class="icon"
            >
              <i
                class="fab fa-twitter"
              />
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`AboutCard renders correctly with mocked query 1`] = `
<body>
  <div>
    <div
      class="card"
    >
      <div
        class="card-image"
      >
        <figure
          class="image"
        >
          <img />
        </figure>
      </div>
      <div
        class="card-content"
      >
        <div
          class="media"
        >
          <div
            class="media-content"
          >
            <p
              class="title is-4"
            >
              Carl B. Smiley
            </p>
            <p
              class="subtitle is-6"
            >
              @linux_pangolin
            </p>
          </div>
        </div>
        <div
          class="level is-mobile"
        >
          <a
            aria-label="GitHub"
            class="level-item"
            href="https://github.com/beeryt"
          >
            <span
              class="icon"
            >
              <i
                class="fab fa-github"
              />
            </span>
          </a>
          <a
            aria-label="LinkedIn"
            class="level-item"
            href="https://www.linkedin.com/in/carl-b-smiley"
          >
            <span
              class="icon"
            >
              <i
                class="fab fa-linkedin"
              />
            </span>
          </a>
          <a
            aria-label="StackExchange"
            class="level-item"
            href="https://stackexchange.com/users/13595322/linux-pangolin"
          >
            <span
              class="icon"
            >
              <i
                class="fab fa-stack-overflow"
              />
            </span>
          </a>
          <a
            aria-label="Twitter"
            class="level-item"
            href="https://twitter.com/linux_pangolin"
          >
            <span
              class="icon"
            >
              <i
                class="fab fa-twitter"
              />
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</body>
`;
